lib/formatCommitMessage.js

```diff
@@ -1,65 +1,32 @@
 /* eslint-disable complexity */
 
-const wrap = require('word-wrap');
-
-const MAX_LINE_WIDTH = 72;
-
-const makeAffectsLine = function (answers) {
-  const selectedPackages = answers.packages;
-
-  if (selectedPackages && selectedPackages.length) {
-    return `\naffects: ${selectedPackages.join(', ')}`;
-  }
-
-  return '';
-};
-
 const formatCommitMessage = (state) => {
   const {config, answers} = state;
-  const wrapOptions = {
-    indent: '',
-    trim: true,
-    width: MAX_LINE_WIDTH
-  };
-  let head = '';
-  let scope = '';
+  const scope = !answers.scope || answers.scope === 'none' ? '' : `(${answers.scope})`;
+  const scopeEmoji = config.disableEmoji || !config.types[answers.type].emoji ? '' : `${config.types[answers.type].emoji} `;
+  const body = (answers.body || '').trim();
+  const breaking = (answers.breaking || '').trim();
+  const issues = (answers.issues || '').trim();
+  const msg = [`${answers.type}${scope}: ${scopeEmoji}${answers.subject}`];
 
-  if (answers.scope && answers.scope !== 'none') {
-    scope = `(${answers.scope})`;
-  }
-
-  if (config.disableEmoji) {
-    head = answers.type + scope + ': ' + answers.subject;
-  } else {
-    const emoji = config.types[answers.type].emoji;
-    const emojiPrefix = emoji ? emoji + ' ' : '';
-
-    head = answers.type + scope + ': ' + emojiPrefix + answers.subject;
+  if (body) {
+    msg.push(body);
   }
 
-  const affectsLine = makeAffectsLine(answers);
-
-  // Wrap these lines at MAX_LINE_WIDTH character
-  const body = wrap((answers.body || '') + affectsLine, wrapOptions);
-  const breaking = wrap(answers.breaking, wrapOptions);
-  const issues = wrap(answers.issues, wrapOptions);
```

Comment:
I removed the word wrapping because it too was causing truncation. If the long description was long enough to cause a wrap, everything after that wrap (including affects, breaking changes, and issues) would never come through on the commit message.

Unfortunately, I didn't get any responses back from others about what system they are on. But I am on Windows, though this wasn't always an issue, so I don't know what changed that caused this breaking change.

Reply:
Maybe word wrapping truncated lines for the same reason &mdash; on Windows it expects different line breaks.

Reply:
Since there's no specification (that I saw) in the length of lines in commit messages, I felt it was safe to remove. The other option would be to figure out some sort either cross platform solution or introduce some sort of OS detection.

But I think it's a bad idea for us to guess how to solve it, since we haven't actually identified if it's an OS issue, a terminal issue, or if it spans multiple.

If this were website development, I'd advocate for feature detection rather than platform detection. But again, I don't think that's going to be simple or possible

Do you want to go with this solution for now? And then you can follow up and try to solve it however you think is best when you have more time?

Reply:
[word-wrap](https://www.npmjs.com/package/word-wrap#optionsnewline) does provide a way to change the newline characters, but I don't have a way to test this cross platform. Considering the number of OSes and terminals, testing all of the combinations to identify the issue and validate the solution is going to be challenging

Reply:
Maybe we can wrap the lines as before, but feed them through `-m` so Git itself figures out how to format them?

Something like this:

```js
const body = wrap((answers.body || '') + affectsLine, wrapOptions);
msg.push(...body.split('\n'));
```

Reply:
This will cause them to appear as separate "paragraphs" (with a blank line between each) instead of adjacent content

Reply:
Is there a specific use case the word wrapping is solving for? Since it's not a commitlint spec, and the messages wrap by default in the terminals that I'm aware of as well as on the Git websites, I'm not sure what scenario or tool this is necessary for.

Reply:
It is a widely used convention that Git commit messages should be wrapped at 72 characters. It is necessary for text to look nice on the screen.

Reply:
I have tried a whole bunch of options:
[https://github.com/ArrayKnight/git-cz/pull/1](https://github.com/ArrayKnight/git-cz/pull/1)

But nothing works properly. I've tried:

- '\n'
- '\r'
- '\r\n'
- '\n\r'
- ASCII and Unicode variants of the above
- Github's pretty format placeholder ('%n')
- Using the `--cleanup=verbatim` [flag](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---cleanupltmodegt)

The offending character is the newline ('\n') which causes truncation of the entire rest of the message.

I really think that this fix needs to go live without word wrapping or nobody on Windows is going to be able to use your tool.

Reply:
@streamich Any thoughts on how this can progress?

Reply:
@rodrigograca31 Any chance you'd be able to weigh in on this?

Reply:
@ArrayKnight I've taken a quick look at this PR before and stayed away because it looked complex and @streamich was already on it. and I also wanted to enable multiple lines somehow but this was already underway...
Sorry. For now, I will let you guys find a solution...

Reply:
I'm confused I see so many changes for something that **seems simple**

@ArrayKnight do you have zoom or something else?

Reply:
I spent the last 2 hours making a Windows VM, installed node, git etc and was able to reproduce the bug but wasnt able to solve it yet. Its definitely something weird with `Enter` ... 🤔 if I do `--dry-run` and copy and paste the output it works but if I tell it to run directly the bug shows up .... also used `require("os").EOL` instead of `\n\n` with no good results. the best I got was removing all `\n\n` but that is not realistic.

Reply:
Im pretty sure I fixed it on windows. but now on Linux the `shellescape` function is turning:
`git commit --message "abc" --message "" --message "def"`
into
`git commit --message "abc" --message --message "def"` (escaping "")

and then I get `error: pathspec 'def' did not match any file(s) known to git`

Reply:
The downside of using the `--message` option to break lines is that there will actually be an extra empty line between every line. Git's '-m' treats every line as a separate paragraph which works nicely for the sections of the message (body, affects, issues, breaking, etc).

However, this is going to be less than ideal when trying to use it for the word wrapping since this will be each sentence that causes wrapping will get broken into multiple paragraphs.

Reply:
Exactly. I'm not sure how to move forward... 🤔🤔

Reply:
could we use the commit file? 🤔
write to the file, hopefully with linebreaks, and tell git to commit...

Reply:
I also say this:

`git commit -m $'first line\nsecond line'` as a possible solution but I wasn't sure how we can add `$` before the string.... 🤔🤔🤔

Reply:
This string interpolation (?) `-m $'...'` idea seems clever, as long as it works on all platforms.

If that doesn't work, we could try using the file as a backup